cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -478,7 +484,15 @@ protected Future<Void> maybeRollKafka(
                     kafka.getKafkaVersion(),
                     allowReconfiguration,
                     eventsPublisher
-            ).rollingRestart(podNeedsRestart);
+            ).rollingRestart(podNeedsRestart)
+                .compose(v -> Future.succeededFuture(), throwable -> {
+                    // an invalid configuration warning for the user, but it doesn't fail the reconciliation
+                    if (throwable instanceof KafkaRoller.InvalidConfiguration) {
+                        kafkaStatus.addCondition(StatusUtils.buildWarningCondition("InvalidConfiguration", throwable.getMessage()));
+                        return Future.succeededFuture();
```

Comment:
can I ask why we don't fail the reconciliation? Technically we failed to reconcile the configuration update due to an invalid value? If we don't fail it, would it be obvious to the user to notice that they need to fix the configuration?